Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift

```diff
@@ -0,0 +1,270 @@
+            let signature = try self.sign(
+                archivePath: archivePath,
+                signatureFormat: self.signatureFormat,
+                signingIdentity: self.signingIdentity,
+                privateKeyPath: self.privateKeyPath,
+                observabilityScope: swiftTool.observabilityScope
+            )
+
+            // step 4: publish the package
+            swiftTool.observabilityScope
+                .emit(info: "publishing '\(self.packageIdentity)' archive at '\(archivePath)' to '\(registryURL)'")
+            try self.publish(
+                packageIdentity: self.packageIdentity,
+                registryURL: registryURL,
+                archivePath: archivePath,
+                signature: signature,
+                registryClient: registryClient,
+                observabilityScope: swiftTool.observabilityScope
+            )
+        }
+
+        func getPublishRequirements(
+            registryURL: URL,
+            registryClient: RegistryClient,
+            observabilityScope: ObservabilityScope
+        ) throws -> RegistryClient.PublishRequirements {
+            try tsc_await { callback in
+                registryClient.getPublishRequirements(
+                    registryURL: registryURL,
+                    observabilityScope: observabilityScope,
+                    callbackQueue: .sharedConcurrent,
+                    completion: callback
+                )
+            }
+        }
+
+        func archiveSource(
+            packageIdentity: PackageIdentity,
+            packageDirectory: AbsolutePath,
+            metadataPath: AbsolutePath?,
+            customWorkingDirectory: AbsolutePath?,
+            cancellator: Cancellator?,
+            observabilityScope: ObservabilityScope
+        ) throws -> AbsolutePath {
+            let workingDirectory = customWorkingDirectory ?? Workspace.DefaultLocations
+                .scratchDirectory(forRootPackage: packageDirectory).appending(components: ["registry", "publish"])
+
+            let archivePath = workingDirectory.appending(component: "\(packageIdentity).zip")
+
+            // create temp location for sources
+            let sourceDirectory = workingDirectory.appending(components: "source", "\(packageIdentity)")
+            try localFileSystem.createDirectory(sourceDirectory, recursive: true)
+
+            // TODO: filter other unnecessary files, and/or .swiftpmignore file
+            let ignoredContent = [".build", ".git"]
+            let packageContent = try localFileSystem.getDirectoryContents(packageDirectory)
+            for item in (packageContent.filter { !ignoredContent.contains($0) }) {
+                try localFileSystem.copy(
+                    from: packageDirectory.appending(component: item),
+                    to: sourceDirectory.appending(component: item)
+                )
+            }
+
+            // include metadata from non-standard location in the archive
```

Comment:
This is doing more than what the spec says--custom metadata file location is used for `in-request` metadata only; it doesn't say SwiftPM would copy from custom metadata file location to the archive. 

I don't have a strong opinion, just need to update the proposal accordingly.

Reply:
thanks for pointing this our. this feels like the more intuitive behavior to me.